### DISCUSSION TITLE
A snap for LEGO

### DISCUSSION THREAD

**Discussion by gruyaume** [go-acme/lego#2133] [Show and tell]
# A snap for LEGO

Hello, I wrote a snap for LEGO, making it easy to install and use LEGO on any Linux distribution that supports snaps.

## Usage

```bash
sudo snap install lego
sudo lego \
  --email="you@example.com" \
  --domains="example.com" \
  --server=https://acme-staging-v02.api.letsencrypt.org/directory \
  --http \
  --http.port :8080 \
  run
```

**Note**: The LEGO snap can only write to the `/var/snap/lego/common/.lego` directory.

## Architectures

The snap is available for the following architectures:
- amd64
- arm64
- riscv64
- armhf
- s390x
- ppc64el

## Confinement

The snap is strictly confined, meaning it only has access to what it requires to work and for LEGO that means [network binding](https://snapcraft.io/docs/network-bind-interface). 

## Documentation

My goal is for this snap to be documented in the official LEGO documentation. I will create a PR for this soon.

## Maintenance

Right now I maintain the snap, I would be happy to bring it back to the upstream project here if maintainers of LEGO prefer to maintain the snap in the same place.

## Reference

- https://snapcraft.io/lego
- https://github.com/gruyaume/lego-snap/

## Issues

If you have any issue with the snap, please open an issue [here](https://github.com/gruyaume/lego-snap/issues).

**Comment by ldez** [comment]
Hello,

Thank you.

If you can do the integration inside of goreleaser workflow, it can be integrated inside the lego repo.

- https://goreleaser.com/customization/snapcraft/
- https://github.com/go-acme/lego/blob/master/.goreleaser.yml
